library/ssl_srv.c

```diff
@@ -3585,15 +3585,19 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl )
     /* Read the message without adding it to the checksum */
     do {
 
-        if( ( ret = mbedtls_ssl_read_record_layer( ssl ) ) != 0 )
+        do ret = mbedtls_ssl_read_record_layer( ssl );
+        while( ret == MBEDTLS_ERR_SSL_CONTINUE_PROCESSING );
```

Comment:
@hanno-arm but with current implementation, if `mbedtls_ssl_parse_record_header` fails, the error will be returned from `mbedtls_ssl_read_record_layer ` to `ssl_parse_certificate_verify`, which will fail and return error to user. Am I missing something?